sympy/solvers/solvers.py

```diff
@@ -1061,6 +1061,20 @@ def _sympified_list(w):
         # this mod the return value is []
         ok = False
         if fi.free_symbols & symset:
+            if isinstance(fi, Expr):
+                print('this is fi',fi, type(fi))
+                if fi.is_constant():
+                    # if it's constant and is nonzero, it invalidates
+                    # any solution
+                    if fi.equals(0):
+                        # any symbols that appear only in this equation
+                        # need not be considered
+                        fisyms = fi.free_symbols & symset
+                        for _ in f:
+                            if _ != fi:
+                                fisyms -= _.free_symbols
+                        symset -= fisyms
+                        continue
```

Comment:
Although an equation might simplify to zero if it involves the unknowns then it might still invalidate the solutions to the other equations e.g.:
```
In [1]: solve([x*(x-1), 1/x - 1/(x - 1) + 1/(x*(x - 1))], [x])
this is fi x*(x - 1) <class 'sympy.core.mul.Mul'>
this is fi -1/(x - 1) + 1/x + 1/(x*(x - 1)) <class 'sympy.core.add.Add'>
Out[1]: [(0,), (1,)]
```

Reply:
Now handled.